src/cpp/core/r_util/RTokenizer.cpp

```diff
@@ -556,7 +556,7 @@ wchar_t RTokenizer::peek()
 
 wchar_t RTokenizer::peek(std::size_t lookahead)
 {
-   if ((pos_ + lookahead) >= data_.end())
+   if (lookahead < data_.size() && pos_ >= data_.end() - lookahead)
```

Comment:
```suggestion
   if ((lookahead < data_.size()) && (pos_ >= data_.end() - lookahead))
```

Reply:
Double checking here -- should this be 
`(lookahead < data_.size()) || (pos_ >= data_.end() - lookahead)`